glow/event_log.go

```diff
@@ -0,0 +1,112 @@
+package glow
+
+// This file contains a simple event logger, designed to collect a limited
+// number of timestamped logs and dump them to file or terminal.
+
+import (
+	"container/list"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+)
+
+type event struct {
+	ts   time.Time
+	line string
+}
+
+// Options for the event log.
+type EventLogOptions struct {
+	Expiry   time.Duration // Older logs will be removed, 0 means never expire
+	MaxCount int           // Limit the number of logs to collect, 0 means unlimited
+}
+
+type EventLog struct {
+	Options EventLogOptions
+	Logs    *list.List
+	mu      sync.Mutex
+}
+
+// Create a new event log.
+// The default behavior is to never expire, and keep unlimited logs.
```

Comment:
default behavior should probably be like 100,000 log lines or something

Reply:
allowing anything to be unlimited is just bad habit

Reply:
I think I will limit via memory rather than lines